### PR TITLE
feat: allow more decorators in ethante

### PR DESCRIPTION
**Pull request by trinitys7**
Right now we only use ethante in rollapp-evm. But we also want to add some decorators for the cosmos transaction. This PR will allow ethante to add more decorators

### CODE REVIEW

File: app/ante/ante.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ const (
 // Ethereum or SDK transaction to an internal ante handler for performing
 // transaction-level processing (e.g. fee payment, signature verification) before
 // being passed onto it's respective handler.
-func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
+func NewAnteHandler(options HandlerOptions, anteDecorators []sdk.AnteDecorator) (sdk.AnteHandler, error) {
 	if err := options.validate(); err != nil {
 		return nil, err
 	}
```
Comment (danwt) [inline app/ante/ante.go]: can you add as an option instead

```
// HandlerOptions extend the SDK's AnteHandler options by requiring the IBC
// channel keeper, EVM Keeper and Fee Market Keeper.
type HandlerOptions struct {
...
	ExtraCosmosAnteDecorators []sdk.AnteDecorator
}
```

File: app/ante/eip712.go [hunk 2]
```diff
@@ -50,8 +50,8 @@ func init() {
 
 // Deprecated: NewLegacyCosmosAnteHandlerEip712 creates an AnteHandler to process legacy EIP-712
 // transactions, as defined by the presence of an ExtensionOptionsWeb3Tx extension.
-func NewLegacyCosmosAnteHandlerEip712(options HandlerOptions) sdk.AnteHandler {
-	return sdk.ChainAnteDecorators(
+func NewLegacyCosmosAnteHandlerEip712(options HandlerOptions, anteDecorators []sdk.AnteDecorator) sdk.AnteHandler {
+	anteDecorators = append(anteDecorators,
```
Comment (danwt) [inline app/ante/eip712.go:54]: yeah I'm not sure if it should go first 🤔 
let me think

File: app/ante/handler_options.go [hunk 3]
```diff
@@ -83,8 +83,8 @@ func newEthAnteHandler(options HandlerOptions) sdk.AnteHandler {
 	)
 }
 
-func newCosmosAnteHandler(options HandlerOptions) sdk.AnteHandler {
-	return sdk.ChainAnteDecorators(
+func newCosmosAnteHandler(options HandlerOptions, anteDecorators []sdk.AnteDecorator) sdk.AnteHandler {
+	anteDecorators = append(anteDecorators,
```
Comment (danwt) [inline app/ante/handler_options.go:88]: yeah I'm not sure if it should go first 🤔 
let me think

**Review by danwt — CHANGES_REQUESTED** [review]

**Review by danwt — COMMENTED** [review]
you dont need to specify  anteDecorators []sdk.AnteDecorator as argument,
Just read it from the options

**Review by mtsitrin — APPROVED** [review]

**Comment by danwt** [comment]
will take another approach

**Pull request closed without merging**